### PR TITLE
Add Service Layer Guidelines

### PR DESCRIPTION
**Pull request by Januson**

### CODE REVIEW

File: docs/candlepin/service_layer_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    @Verify(ActivationKey.class)
+    @NotEmpty String activationKeyId,
+    @NotNull ActivationKeyDTO update) {
+    log.debug("Updating activation key: {}", activationKeyId);
+    ActivationKey toUpdate = this.activationKeyController
+        .updateActivationKey(activationKeyId, toActivationKey(update));
+    return this.translator.translate(toUpdate, ActivationKeyDTO.class);
+}
+```
+Translate DTO to appropriate entity. This is just one to one translation as input validations are done by bean validation at the endpoint entry and business validation will be done by the controller.
```
Comment (nikosmoum) [inline docs/candlepin/service_layer_guidelines.md]: Now that we've agreed to use the Translator framework, it's better to update this (and the code section below. no need to write the full translator, just how we'd call it from the resource) to reflect it.

**Comment by nikosmoum** [comment]
needs rebase to master to get the travis build passing

File: docs/candlepin/service_layer_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+---
+title: Service Layer Guidelines
+---
+{% include toc.md %}
+
+# Service Layer Guidelines
+
```
[the remark this answers is not in the document]
Comment (Januson) [reply docs/candlepin/service_layer_guidelines.md]: Done.

File: docs/candlepin/service_layer_guidelines.md (hunk 1 above)
Comment (Januson) [reply docs/candlepin/service_layer_guidelines.md]: Done.

File: docs/candlepin/service_layer_guidelines.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+## Resources
+* Any logic unrelated to the presentation layer should be extracted into a Controller or other classes hidden behind controller
+* Should not other resource methods. It should call Controller instead
```
[the remark this answers is not in the document]
Comment (Januson) [reply docs/candlepin/service_layer_guidelines.md]: Done.

File: _data/toc.yaml [hunk 4]
```diff
@@ -90,6 +90,7 @@
           - section: javascript_rules
           - section: dtos
           - section: authentication_and_authorization
+          - section: service_layer_guidelines
```
[the remark this answers is not in the document]
Comment (Januson) [reply _data/toc.yaml]: Done.

**Pull request merged**